### PR TITLE
Improved logging for store copy / Updated store info to handle multi-arch images

### PR DESCRIPTION
**Pull request by amartin120**
**Please check below, if the PR fulfills these requirements:**
- [x] The commit message follows the guidelines.
- [ ] Tests for the changes have been added (for bug fixes / features).
- [ ] Docs have been added / updated (for bug fixes / features).

**What kind of change does this PR introduce?**
* Improved logging for store copy 
* Updated store info to handle multi-arch images

**What is the current behavior?**
* `hauler store copy` provides no incremental feedback.
* `hauler store info` is missing data for the majority of images.

**What is the new behavior (if this is a feature change)?**
* `hauler store copy` provides better logging.
```
❯ ./bin/hauler store copy registry://your.registry.com                                                                                                                                                                                                              
11:31AM INF carbide/k3s-docs:0.1.2
11:31AM INF rancher/mirrored-library-traefik:2.9.1
11:31AM INF hauler/sha256sum-amd64.txt:latest
11:31AM INF carbide/heimdall2:0.1.1
11:31AM INF carbide/carbide-whitelabel:0.1.1
11:31AM INF rancher/hardened-calico:v3.26.1-build20230802
11:31AM INF carbide/stigatron-hook:0.2.0
11:31AM INF copied artifacts to [your.registry.com]
```

* `hauler store info` now handles multi-arch images.

![image](https://github.com/rancherfederal/hauler/assets/42001113/ef69dcfa-b32e-4d07-ac61-01cf30a49af1)


**Does this PR introduce a breaking change?**
* No

**Other information**:
* <!-- Any additional information -->

### CODE REVIEW

File: cmd/hauler/cli/store/info.go [hunk 1]
```diff
@@ -4,8 +4,9 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
-	"strings"
-	"text/tabwriter"
+	"github.com/olekukonko/tablewriter"
```
Comment (dweomer) [inline cmd/hauler/cli/store/info.go:7]: MIT license, okay good.

Comment (amartin120) [reply cmd/hauler/cli/store/info.go:7]: I'm a fan of his tablewriter.   I've used it in a few projects now.

File: cmd/hauler/cli/store/info.go [hunk 2]
```diff
@@ -44,49 +45,109 @@ func InfoCmd(ctx context.Context, o *InfoOpts, s *store.Layout) error {
...
-		var m ocispec.Manifest
-		if err := json.NewDecoder(rc).Decode(&m); err != nil {
-			return err
-		}
-		i := newItem(s, desc, m)
-		var emptyItem item
-		if i != emptyItem {
-			items = append(items, i)
+		// handle multi-arch images 
+		if desc.MediaType == consts.OCIImageIndexSchema || desc.MediaType == consts.DockerManifestListSchema2 {
```
Comment (dweomer) [inline cmd/hauler/cli/store/info.go:49]: I imagine that these media type comparisons/checks will be with us for some years. I wonder if longer term than this particular change-set if it makes sense to unify them conceptually? i.e. for this one, possibly something like `hauler.IsImageIndex()` / `hauler.IsImageShema()`

Comment (amartin120) [reply cmd/hauler/cli/store/info.go:49]: I like that idea.

Comment (dweomer) [reply cmd/hauler/cli/store/info.go:49]: The idea isn't for saving characters/lines of code so much as conceptual concision.

**Review by dweomer — APPROVED** [review]

**Pull request merged**